### PR TITLE
feat: enable multi arch build

### PR DESCRIPTION
**Pull request by barnabasbusa**

### CODE REVIEW

**Comment by barnabasbusa** [comment]
This PR will enable the repo to push a multi arch image to dockerhub. The build takes approx 7:30mins vs 1 min, but as this job isn't ran that frequently, I think its a fair compromise.

File: .github/workflows/test-build-push.yaml [hunk 1]
```diff
@@ -16,11 +16,11 @@ jobs:
...
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
     - name: Set up Go
-      uses: actions/setup-go@v2
+      uses: actions/setup-go@v5
       with:
-        go-version: 1.20.5
+        go-version: 1.21.5
```
Comment (parithosh) [inline .github/workflows/test-build-push.yaml:23]: the project is still based on go 1.20, so you shouldn't be using 1.21 for it. 

https://github.com/MariusVanDerWijden/tx-fuzz/blob/master/go.mod#L3C1-L3C8

Comment (MariusVanDerWijden) [reply .github/workflows/test-build-push.yaml:23]: You can just move the go.mod to 1.21 as well

Comment (MariusVanDerWijden) [reply .github/workflows/test-build-push.yaml:23]: but 1.21.5 is not a stable version, I'd rather stay on 1.20 or move to 1.21.6

Comment (barnabasbusa) [reply .github/workflows/test-build-push.yaml]: reverted to 1.20.5, let this be updated in a separate PR.

**Pull request merged**